### ISSUE TITLE
New Pull Request Process

### ISSUE THREAD

**Issue by freddydk** [microsoft/AL-Go#252]
### Summary

Previously, the CI/CD workflow was setup to trigger on Pull Requests, causing GitHub to automatically add all jobs in the CI/CD workflow as checks in the Pull Request. This had some disadvantages:

- Fork Pull Requests does not have access to secrets and cannot run builds if a license or an insider sas token is required
- Fork Pull Requests might contain malicious code which (if not discovered by the person accepting the PR) will run during the workflow
- All jobs in the CI/CD workflow are added as checks, also the initialization job etc. - doesn't look good

Current PR process looks like this on a drawing:
![Image](https://user-images.githubusercontent.com/10775043/196020645-ebccc254-eff3-4486-a36f-1a3115eb41fa.png)

### Intended Outcome

We need to prevent malicious fork pull requests to rely on a human to discover the thread

### How will it work?

The solution is to create a new workflow, which catches the Pull Request, packs all changed files and uploads these to a short-lived artifact.
The CI/CD workflow will start at the same time – running without the files from the PR – it will download the changed files, investigate for malicious files and if everything is OK – run the CI/CD with secrets and all
On a drawing, it looks like this:
![Image](https://user-images.githubusercontent.com/10775043/196020660-bdcbca97-85ef-4a72-af1e-f300e1a9a971.png)
Mechanism described here:  [Keeping your GitHub Actions and workflows secure Part 1: Preventing pwn requests | GitHub Security Lab](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/)

The CI/CD workflow will add itself to the checks and all projects will add themselves as well.
All checks need to be completed before the PR can be approved.
